addons/mail/static/src/models/dropdown_menu_view.js

```diff
@@ -0,0 +1,181 @@
+         * considered together.
+         */
+        setNextActive() {
+            const activeIndex = this.items.findIndex(
+                item => item === this.activeItem
+            );
+            if (activeIndex === this.items.length - 1) {
+                // loop when reaching the end of the list
+                this.setFirstActive();
+                return;
+            }
+            const nextDropdownItemView = this.items[activeIndex + 1];
+            this.update({ userExplicitSelection: nextDropdownItemView });
+        },
+        /**
+         * Sets the previous dropdownItem as active. Main and extra records are
+         * considered together.
+         */
+        setPreviousActive() {
+            const activeIndex = this.items.findIndex(
+                item => item === this.activeItem
+            );
+            if (activeIndex === 0) {
+                // loop when reaching the start of the list
+                this.setLastActive();
+                return;
+            }
+            const previousDropdownItemView = this.items[activeIndex - 1];
+            this.update({ userExplicitSelection: previousDropdownItemView });
+        },
+        /**
+         * Closes the popover when clicking outside, if appropriate.
+         *
+         * @private
+         * @param {MouseEvent} ev
+         */
+        _onClickCaptureGlobal(ev) {
+            if (!this.component || !this.component.root.el) {
+                return;
+            }
+            if (this.anchorRef && this.anchorRef.el && this.anchorRef.el.contains(ev.target)) {
+                return;
+            }
+            if (this.component.root.el.contains(ev.target)) {
+                return;
+            }
+            this.delete();
+        },
+    },
+    fields: {
+        activeItem: one('DropdownItemView', {
+            compute() {
+                if (this.items.includes(this.userExplicitSelection)) {
+                    return this.userExplicitSelection;
+                }
+                const firstDropdownItemView = this.items[0];
+                return firstDropdownItemView;
+            },
+        }),
+        anchorRef: attr({
+            compute() {
+                if (this.ownerAsComposerSuggestion) {
+                    return this.ownerAsComposerSuggestion.textInput.textareaRef;
+                }
+                return clear();
+            },
+            required: true,
+        }),
+        component: attr(),
+        extraItems: many('DropdownMenuExtraItemView', {
+            compute() {
+                if (this.ownerAsComposerSuggestion) {
+                    return this.ownerAsComposerSuggestion.extraSuggestions.map(suggestable => ({ suggestable }));
+                }
+                return clear();
+            },
+            inverse: 'dropdownMenuViewOwner',
+        }),
+        hasToScrollToActiveItem: attr({
+            default: false,
+        }),
+        items: many('DropdownItemView', {
```

Comment:
similarly confusing field name, and especially also model name

If `DropdownItemView` is a part of `DropdownMenuView`, it would make more sense as `DropdownMenuViewItem` or `DropdownMenuView.Item`